File: web/src/render/renderSelfClosedTag.ts
import type MarkdownNode from "../types/MarkdownNode";
import type RendererState from "../types/RendererState";
import { endNewLine, startNewLine } from "./utils";

export default function render(node: MarkdownNode, state: RendererState, tag: string): void {
	startNewLine(node, state);
	state.output += `<${tag} />`;
	endNewLine(node, state);
}

File: dotnet/Allmark/Render/RenderSelfClosedTag.cs
namespace Allmark.Render;

using Allmark.Types;

public static class RenderSelfClosedTag
{
	public static void Execute(MarkdownNode node, RendererState state, string tag)
	{
		RenderUtils.StartNewLine(node, state);
		state.Output.Append($"<{tag} />");
		RenderUtils.EndNewLine(node, state);
	}
}
